### ISSUE TITLE
Display hostname in confirmation message when sharing eval to promptfoo server

### ISSUE THREAD

**Issue by efung** [promptfoo/promptfoo#1182]
Using the promptfoo CLI, if I'm sharing the latest eval to a self-hosted server (following [these instructions](https://www.promptfoo.dev/docs/usage/self-hosting#pointing-the-promptfoo-client-to-your-hosted-instance)), it would be nice if the confirmation message included the hostname it was sharing to.

That is, change this message:

```
Create a private shareable URL of your most recent eval?

To proceed, please confirm [Y/n]
```

to something like:

```
Create a private shareable URL of your most recent eval on promptfoo.example.com?

To proceed, please confirm [Y/n]
```

Since the hostname is set via an environment variable, having this confirmation would re-assure someone that they are indeed sending it to their instance, and not the public one.

**Comment by typpo** [comment]
Good suggestion!  Done in #1183

**Comment by mldangelo** [comment]
Thanks so much for this suggestion, @efung, and for all your other helpful suggestions. We released a fix for this in [0.72.1](https://github.com/promptfoo/promptfoo/releases/tag/0.72.1).
